### PR TITLE
[trel-dnssd] parse `xa` TXT value as binary

### PR DESCRIPTION
**Pull request by simonlingoogle**
Depends-On: openthread/openthread#7401

### CODE REVIEW

**Review by abtink — APPROVED** [review]
LGTM. 👍

File: src/trel_dnssd/trel_dnssd.cpp [hunk 1]
```diff
@@ -478,14 +478,9 @@ void TrelDnssd::Peer::ReadExtAddrFromTxtData(void)
     {
         if (StringUtils::EqualCaseInsensitive(txtEntry.mName, kTxtRecordExtAddressKey))
         {
-            char extAddrHexBuf[sizeof(mExtAddr) * 2 + 1];
+            VerifyOrExit(txtEntry.mValue.size() == sizeof(mExtAddr));
```
Comment (bukepo) [inline src/trel_dnssd/trel_dnssd.cpp:481]: should we add a debug level log here?

Comment (simonlingoogle) [reply src/trel_dnssd/trel_dnssd.cpp:481]: Make sense. Will add.

Comment (simonlingoogle) [reply src/trel_dnssd/trel_dnssd.cpp:481]: Added a `INFO` log if we can't dissect Extended Address from TXT data. 
BTW, this error should not happen in practise.

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**